core_api/src/routes/chat.py

```diff
@@ -79,36 +79,41 @@ async def rag_chat_streamed(
     request = await websocket.receive_text()
     chat_request = ChatRequest.model_validate_json(request)
 
-    selected_chain, params = await semantic_router_to_chain(chat_request, user_uuid, routable_chains, route_layer)
+    selected_chain, params, route = await semantic_router_to_chain(
+        chat_request, user_uuid, routable_chains, route_layer
+    )
 
     async for event in selected_chain.astream_events(params.model_dump(), version="v1"):
         kind = event["event"]
+        log.info("%s %s %s", kind, route, event["data"])
         if kind == "on_chat_model_stream":
-            await websocket.send_json({"resource_type": "text", "data": event["data"]["chunk"].content})
+            await websocket.send_json({"resource_type": "text", "data": event["data"]["chunk"].content, "route": route})
```

Comment:
Does every bit of text need the route? It's one route for the entire response, right?

Reply:
fair point. killing this PR in anycase